Basic/DyanmicPrefabs/Assets/Scripts/NetworkManagerHud.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Runtime.CompilerServices;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+
+[RequireComponent(typeof(NetworkManager))]
+[DisallowMultipleComponent]
+public class NetworkManagerHud : MonoBehaviour
+{
+    NetworkManager m_NetworkManager;
+
+    UnityTransport m_Transport;
+
+    GUIStyle m_LabelTextStyle;
+
+    // This is needed to make the port field more convenient. GUILayout.TextField is very limited and we want to be able to clear the field entirely so we can't cache this as ushort.
+    string m_PortString = "7777";
+    string m_ConnectAddress = "127.0.0.1";
+
+    public Vector2 DrawOffset = new Vector2(10, 10);
+
+    public Color LabelColor = Color.black;
+
+    void Awake()
+    {
+        // Only cache networking manager but not transport here because transport could change anytime.
+        m_NetworkManager = GetComponent<NetworkManager>();
+        m_LabelTextStyle = new GUIStyle(GUIStyle.none);
+    }
+
+    void OnGUI()
+    {
+        m_LabelTextStyle.normal.textColor = LabelColor;
+
+        m_Transport = (UnityTransport)m_NetworkManager.NetworkConfig.NetworkTransport;
+
+        GUILayout.BeginArea(new Rect(DrawOffset, new Vector2(200, 200)));
+
+        if (IsRunning(m_NetworkManager))
+        {
+            DrawStatusGUI();
+        }
+        else
+        {
+            DrawConnectGUI();
+        }
+
+        GUILayout.EndArea();
+    }
+
+    void DrawConnectGUI()
+    {
+        GUILayout.BeginHorizontal();
+        GUILayout.Space(10);
+        GUILayout.Label("Address", m_LabelTextStyle);
+        GUILayout.Label("Port", m_LabelTextStyle);
+
+        GUILayout.EndHorizontal();
+
+        GUILayout.BeginHorizontal();
+
+        m_ConnectAddress = GUILayout.TextField(m_ConnectAddress);
+        m_PortString = GUILayout.TextField(m_PortString);
+        if (ushort.TryParse(m_PortString, out ushort port))
+        {
+            m_Transport.SetConnectionData(m_ConnectAddress, port);
```

Comment:
I don't like doing transport/business logic/dangerous calls every GUI call. Pretty sure this line is harmless, just pointing out a long term risk.

Reply:
Yeaaah, not my class - borrowed it from somewhere. It's only needed for the preloading sample though, could be thrown out in favor of some simple ugui ui.

Reply:
As mentioned above, this class won't exist in the follow up UI PR.